### PR TITLE
as-tree 0.12.0 (new formula)

### PR DESCRIPTION
**Pull request by bosr**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/as-tree.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "BlueOak-1.0.0"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "as-tree", "--version"
```
Comment (carlocab) [inline Formula/as-tree.rb:15]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @bosr.

File: Formula/as-tree.rb (hunk 1 above)
Comment (bosr) [reply Formula/as-tree.rb:15]: Hi thanks, I'm not sure this applies for static binaries like rust and go, but ok!

File: Formula/as-tree.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  license "BlueOak-1.0.0"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_equal ".\n└── file\n", shell_output("echo 'file' | #{bin}/as-tree")
```
Comment (carlocab) [inline Formula/as-tree.rb:15]:
```suggestion
    assert_equal ".\n└── file\n", pipe_output("#{bin}/as-tree", "file")
```

**Review by carlocab — COMMENTED** [review]
The test looks great; thanks. Just one small suggestion.

File: Formula/as-tree.rb (hunk 2 above)
Comment (bosr) [reply Formula/as-tree.rb]: Oh, nice thanks, I did not know that one.

**Review by carlocab — APPROVED** [review]
Thanks, @bosr.

**Pull request closed without merging**